// Copyright Istio Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

package istio.workload_discovery.v1;

import "envoy/config/core/v3/config_source.proto";

option go_package = "istio.io/proxy/discovery_proto";

// Contains the metadata for a single workload instance.
message WorkloadMetadataResource {
  // Unique identifier, typically matches IP address.
  string name = 1;

  // The full name of the workload instance.
  string instance_name = 2;

  // The Kubernetes namespace to which the workload instance belongs.
  string namespace_name = 3;

  // The name of the workload provided by the instance. This is typically the
  // Kubernetes deployment name.
  string workload_name = 5;

  // Canonical name of the workload
  string canonical_name = 7;

  // Canonical revision of the workload
  string canonical_revision = 8;
}

message BootstrapExtension {
  envoy.config.core.v3.ConfigSource config_source = 1;
}
